types/index.d.ts

```diff
@@ -121,7 +121,7 @@ declare namespace Realm {
         inMemory?: boolean;
         schema?: (ObjectClass | ObjectSchema)[];
         schemaVersion?: number;
-        sync?: SyncConfiguration;
+        sync?: boolean | SyncConfiguration;
```

Comment:
Technically `false` shouldn't be a valid value.

```suggestion
        sync?: true | SyncConfiguration;
```